### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by J5lx**
All those warnings bugged me so I decided to take them on. Lots of unused parameters and variables, bogus casts and other stuff. Now all remaining warnings are caused by zlib and QuaZip. I tried to find the most sensible solution for each case, but since I needed to venture into parts of the codebase that are unfamiliar to me it would be great if someone who is familiar with them could check these changes as well.

### CODE REVIEW

File: core_lib/structure/layer.cpp [hunk 1]
```diff
@@ -474,7 +474,7 @@ void Layer::mouseRelease( QMouseEvent* event, int frameNumber )
 
 void Layer::editProperties()
 {
-    QRegExp regex("([\uFFEF-\uFFFF])+");
+    QRegExp regex("([\\xFFEF-\\xFFFF])+");
```
Comment (CandyFace) [inline core_lib/structure/layer.cpp]: Have you tested this? as far as I know when you use \x and unicode characters, then you have to encapsulate them in braces.  So it should be [\x{FFEF}-\x{FFFF}] instead.

Comment (J5lx) [inline core_lib/structure/layer.cpp]: The documentation doesn’t mention braces and it seemed to work fine when testing. What you are probably thinking of is QRegularExpression which uses PCRE, see http://www.pcre.org/current/doc/html/pcre2syntax.html#SEC3. QRegExp uses its own syntax.

Comment (CandyFace) [inline core_lib/structure/layer.cpp]: Yup you're right, I mistakenly thought that they used the same syntax.

**Comment by chchwy** [comment]
Good work! @J5lx

**Pull request merged**